node/nodeHelper.go

```diff
@@ -50,16 +63,7 @@ func CreateNode(
 ) (*Node, error) {
 	prepareOpenTopics(networkComponents.InputAntiFloodHandler(), processComponents.ShardCoordinator())
 
-	peerDenialEvaluator, err := blackList.NewPeerDenialEvaluator(
-		networkComponents.PeerBlackListHandler(),
-		networkComponents.PubKeyCacher(),
-		processComponents.PeerShardMapper(),
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	err = networkComponents.NetworkMessenger().SetPeerDenialEvaluator(peerDenialEvaluator)
+	peerDenialEvaluator, err := preparePeerDenialEvaluators(networkComponents, processComponents)
 	if err != nil {
```

Comment:
so we don't need to return full achive peer denial evaluator and set it when creating NewNode?

Reply:
nope.. the peer denial evaluator at node level is used only to check if a peer is denied, which relays on 2 shared components for both regular denial evaluator and full archive one.. so the response would be the same